lib/msf/ui/console/module_option_tab_completion.rb

```diff
@@ -338,15 +338,22 @@ def option_values_target_addrs(mod)
         # Provide the target ports
         #
         def option_values_target_ports(mod)
-          res = [ ]
-          return res if !framework.db.active
-          return res if !mod.datastore['RHOST']
+          res = []
+          return res unless framework.db.active
+          return res if mod.datastore['RHOST'].nil?
+
+          host_address = nil
+          framework.db.hosts.each do |h|
+            if h.address == mod.datastore['RHOST']
```

Comment:
Does this need the same semantics as the previous ipv6 scope handling?

https://en.wikipedia.org/wiki/IPv6_address#Scoped_literal_IPv6_addresses_(with_zone_index)